colour/colorimetry/generation.py

```diff
@@ -0,0 +1,349 @@
+__email__ = 'colour-science@googlegroups.com'
+__status__ = 'Production'
+
+__all__ = [
+    'constant_spd', 'zeros_spd', 'ones_spd', 'gaussian_spd',
+    'single_led_spd_Ohno2005', 'multi_led_spd_Ohno2005'
+]
+
+
+def constant_spd(k, shape=DEFAULT_SPECTRAL_SHAPE, dtype=DEFAULT_FLOAT_DTYPE):
+    """
+    Returns a spectral power distribution of given spectral shape filled with
+    constant :math:`k` values.
+
+    Parameters
+    ----------
+    k : numeric
+        Constant :math:`k` to fill the spectral power distribution with.
+    shape : SpectralShape, optional
+        Spectral shape used to create the spectral power distribution.
+    dtype : type
+        Data type used for the spectral power distribution.
+
+    Returns
+    -------
+    SpectralPowerDistribution
+        Constant :math:`k` to filled spectral power distribution.
+
+    Notes
+    -----
+    -   By default, the spectral power distribution will use the shape given
+        by :attr:`colour.DEFAULT_SPECTRAL_SHAPE` attribute.
+
+    Examples
+    --------
+    >>> spd = constant_spd(100)
+    >>> spd.shape
+    SpectralShape(360.0, 780.0, 1.0)
+    >>> spd[400]
+    100.0
+    """
+
+    wavelengths = shape.range(dtype)
+    values = np.full(len(wavelengths), k, dtype)
+
+    name = '{0} Constant'.format(k)
+    return SpectralPowerDistribution(
+        values, wavelengths, name=name, dtype=dtype)
+
+
+def zeros_spd(shape=DEFAULT_SPECTRAL_SHAPE):
+    """
+    Returns a spectral power distribution of given spectral shape filled with
+    zeros.
+
+    Parameters
+    ----------
+    shape : SpectralShape, optional
+        Spectral shape used to create the spectral power distribution.
+
+    Returns
+    -------
+    SpectralPowerDistribution
+        Zeros filled spectral power distribution.
+
+    Notes
+    -----
+    -   By default, the spectral power distribution will use the shape given
+        by :attr:`colour.DEFAULT_SPECTRAL_SHAPE` attribute.
+
+    Examples
+    --------
+    >>> spd = zeros_spd()
+    >>> spd.shape
+    SpectralShape(360.0, 780.0, 1.0)
+    >>> spd[400]
+    0.0
+    """
+
+    return constant_spd(0, shape)
+
+
+def ones_spd(shape=DEFAULT_SPECTRAL_SHAPE):
+    """
+    Returns a spectral power distribution of given spectral shape filled with
+    ones.
+
+    Parameters
+    ----------
+    shape : SpectralShape, optional
+        Spectral shape used to create the spectral power distribution.
+
+    Returns
+    -------
+    SpectralPowerDistribution
+        Ones filled spectral power distribution.
+
+    Notes
+    -----
+    -   By default, the spectral power distribution will use the shape given
+        by :attr:`colour.DEFAULT_SPECTRAL_SHAPE` attribute.
+
+    Examples
+    --------
+    >>> spd = ones_spd()
+    >>> spd.shape
+    SpectralShape(360.0, 780.0, 1.0)
+    >>> spd[400]
+    1.0
+    """
+
+    return constant_spd(1, shape)
+
+
+def gaussian_spd(peak_wavelength, fwhm, shape=DEFAULT_SPECTRAL_SHAPE):
```

Comment:
It its Gaussian, should it be specified by the standard Gaussian parameters (mean and std)? Scipy also already uses that format out the box: https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.signal.gaussian.html

Reply:
I can / will add add a `method` argument with for example:

- `Standard`, the default that uses *mean* and *std* (which is how I usually use it btw, and have it implemented in a few notebook.)
- `Ohno (2005)` that uses the current path.